nemo/collections/asr/models/ctc_models.py

```diff
@@ -117,6 +122,25 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
 
         # Adapter modules setup (from ASRAdapterModelMixin)
         self.setup_adapters()
+        self.setup_ipl(model_type="ctc")
+
+    def on_fit_start(self):
+        """
+        Cache datastore manifests for non tarred unlabeled data for pseudo labeling.
+        This function prevents caching audio files at the end of every epoch.
+        """
```

Comment:
About this function.
I am not doing caching of pseudo-labels at the end of each epoch to avoid hangs(with `cache_audio` flag ), that's why I need to do caching at the start of training to make sure that unlabeled data from store will not cause any problems later.
Doing this from dataset will mean to do caching each epoch (as each epoch new part of unlabeled data might be used that has not yet been cached).

Reply:
I will think of a way to implement this without causing mixture of different levels.

Reply:
We should not have to override all classes for IPL caching. Its really not the models responsibility to cache data. Inside dataset, check if data is already cached and then call as an option

Reply:
That's not even the issue.
Problem is that for caching we use multiprocessing pool which hangs during training.
I can avoid it by caching audio files one by one, to do that I can add a flag to `_AudioTextDataset` indicating whether caching is happening for pseudo-labeled dataset.

Would that be OK solution?